### DISCUSSION TITLE
Error: The Edge Function "middleware" is referencing unsupported modules

### DISCUSSION THREAD

**Discussion by jsecanell** [vercel/next.js#58584] [Help] [answered]
### Summary

hello, 

i am trying to deploy a project in vercel and finding the following error: 

Error: The Edge Function "middleware" is referencing unsupported modules:
- next: stream, fs, url, path

and no clue why this is not working. any help will be apreciated

**Comment by icyJoseph** [comment]
Hi,

The Middleware in Next.js does not run on Node.js, so Node.js APIs are not supported. If you are not using them yourself, it is most likely, packages that you use, that do.

Take a look here: https://nextjs.org/docs/app/api-reference/edge#unsupported-apis

**Reply by jsecanell** [reply]
and how can I tell which packages use them?

**Reply by icyJoseph** [reply]
Well, I think it is saying it there there, you must be importing something from `next`, accidentally, could you share your imports? or the middleware code? Pseudo-code is good enough.

**Reply by jsecanell** [reply]
**package.json:** 

`{
  "name": "monkeytours",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "prisma generate && next dev",
    "postinstall": "prisma generate",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@fontsource/roboto": "^5.0.8",
    "@next-auth/prisma-adapter": "^1.0.7",
    "@prisma/client": "^5.3.0",
    "@react-email/components": "^0.0.7",
    "@react-email/render": "^0.0.7",
    "@types/node": "20.5.9",
    "@types/react": "18.2.21",
    "@types/react-dom": "18.2.7",
    "axios": "^1.5.0",
    "bcrypt": "^5.1.1",
    "clsx": "^2.0.0",
    "eslint": "8.48.0",
    "eslint-config-next": "^13.5.5",
    "framer-motion": "^10.16.4",
    "next": "^13.5.5",
    "next-auth": "^4.23.1",
    "next-intl": "^3.0.0-beta.19",
    "react": "^18.2.0",
    "react-date-range": "^1.4.0",
    "react-dom": "^18.2.0",
    "react-email": "^1.9.5",
    "react-hook-form": "^7.46.1",
    "react-hot-toast": "^2.4.1",
    "react-icons": "^4.10.1",
    "react-slideshow-image": "^4.2.1",
    "resend": "^1.1.0",
    "typescript": "5.2.2",
    "zustand": "^4.4.1"
  },
  "devDependencies": {
    "@types/bcrypt": "^5.0.0",
    "@types/react-date-range": "^1.4.7",
    "autoprefixer": "^10.4.15",
    "postcss": "^8.4.29",
    "prisma": "^5.6.0",
    "tailwindcss": "^3.3.3"
  }
}
`
and **middleware.ts**

`import createMiddleware from 'next-intl/middleware';
 
export default createMiddleware({
  // A list of all locales that are supported
  locales: ['en', 'es'],
 
  // If this locale is matched, pathnames work without a prefix (e.g. `/about`)
  defaultLocale: 'es',
  
});
 
export const config = {
  // Skip all paths that should not be internationalized. This example skips
  // certain folders and all pathnames with a dot (e.g. favicon.ico)
  matcher: ['/((?!api|_next|_vercel|.*\\..*).*)']
};`

**Reply by icyJoseph** [reply]
Mmm things do look fine, maybe try to upgrade `next-intl` to 3.1.0, but otherwise, there doesn't seem to be anything wrong 🤔

**Reply by jsecanell** [reply]
it does work locally, but not in vercel prod... will give it a go with the upgrade. 

thanks!

**Reply by icyJoseph** [reply]
Also, make sure your Vercel deploy uses the same Node.js version as you have locally.

**Reply by jsecanell** [reply]
seems locally i do v18.7.0 and is also 18.x in vercel...

**Comment by westmantech** [comment]
I have the same issue trying to deploy a supabase project to production.

**Comment by MichaelJCole** [comment]
Go to your Project Settings > Build & Development Settings, change the framework presets from Others to Next.js, and do a new deployment to work!

**Comment by leerob** [comment, marked as answer]
Good news everyone, **Node.js Middleware support landed in Next.js 15.2** (experimental). This is now ready for testing. For any issues you run into, please open new issues so we can track and address before marking it as stable. Thank you!

https://nextjs.org/blog/next-15-2#nodejs-middleware-experimental
